NameLayerMain/src/vg/civcraft/mc/namelayer/command/commands/UpdateName.java

```diff
@@ -0,0 +1,97 @@
+package vg.civcraft.mc.namelayer.command.commands;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.UUID;
+
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import vg.civcraft.mc.namelayer.NameAPI;
+import vg.civcraft.mc.namelayer.NameLayerPlugin;
+import vg.civcraft.mc.namelayer.command.PlayerCommandMiddle;
+import vg.civcraft.mc.namelayer.misc.NameFetcher;
+
+public class UpdateName extends PlayerCommandMiddle {
+
+	private Map<UUID, String> newNames;
+
+	public UpdateName(String name) {
+		super(name);
+		setIdentifier("nlun");
+		setDescription("Updates your name on this server to the one your minecraft account currently has");
+		setUsage("/nlun [CONFIRM]");
+		setArguments(0, 1);
+		newNames = new TreeMap<UUID, String>();
```

Comment:
This needs to be threadsafe. Just wrap the TreeMap in `Collections.synchronizedSortedMap()`.